src/node_file.cc

```diff
@@ -255,6 +271,124 @@ void FileHandle::Close(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void FileHandle::ReleaseFD(const FunctionCallbackInfo<Value>& args) {
+  FileHandle* fd;
+  ASSIGN_OR_RETURN_UNWRAP(&fd, args.Holder());
+  // Just act as if this FileHandle has been closed.
+  fd->AfterClose();
+}
+
+
+void FileHandle::AfterClose() {
+  closing_ = false;
+  closed_ = true;
+  if (reading_ && !persistent().IsEmpty())
+    EmitRead(UV_EOF);
+}
+
+
+int FileHandle::ReadStart() {
+  if (!IsAlive() || IsClosing())
+    return UV_EOF;
+  reading_ = true;
+
+  if (current_read_ != nullptr)
+    return 0;
+
+  if (read_length_ == 0) {
+    EmitRead(UV_EOF);
+    return 0;
+  }
+
+  Local<Object> wrap_obj = env()->filehandlereadwrap_constructor()
+      ->NewInstance(env()->context()).ToLocalChecked();
+  {
+    AsyncHooks::DefaultTriggerAsyncIdScope trigger_scope(
+        env(), get_async_id());
+    current_read_ = new FileHandleReadWrap(this, wrap_obj);
+  }
+  int64_t recommended_read = 65536;
+  if (read_length_ >= 0 && read_length_ <= recommended_read)
+    recommended_read = read_length_;
+
+  current_read_->buffer = EmitAlloc(recommended_read);
+  current_read_->Dispatched();
+
+  uv_fs_read(env()->event_loop(),
+             current_read_->req(),
+             fd_,
+             &current_read_->buffer,
+             1,
+             read_offset_,
+             [](uv_fs_t* req) {
+    FileHandleReadWrap* req_wrap = FileHandleReadWrap::from_req(req);
```

Comment:
Why not std::unique_ptr? 

`std::unique_ptr<FileHandleReadWrap> req_wrap(FileHandleReadWrap::from_req(req));` 

The `delete` then happens automatically at the end of the block.

Reply:
@fhinkel Done! But mostly because moving towards always using smart pointers is good. In this case, I am not sure that it actually makes the code or ownership flow cleaner (so I’ve added a comment for that).